pyaedt/modules/solutions.py

```diff
@@ -1438,586 +1354,467 @@ def assign_weight(self, a, b, taper="flat", port_cont=0):
+        farfield_data["RealizedGain_Theta"] = real_gain
+        real_gain = 2 * np.pi * np.abs(np.power(farfield_data["rEPhi"], 2)) / incident_power / 377
+        farfield_data["RealizedGain_Phi"] = real_gain
+        farfield_data["Element_Location"] = port_positions
+        return farfield_data
+
+    # fmt: off
     @pyaedt_function_handler()
     def plot_farfield_contour(
-        self,
-        qty_str="RealizedGain",
-        phi_scan=0,
-        theta_scan=0,
-        title="RectangularPlot",
-        convert_to_db=True,
-        export_image_path=None,
+            self,
+            farfield_quantity="RealizedGain",
+            phi_scan=0,
+            theta_scan=0,
+            title="RectangularPlot",
+            format_quantity="dB10",
+            export_image_path=None,
+            levels=64,
+            show=True,
+            **kwargs
     ):
-        """Create a Contour plot of specified quantity.
+        # fmt: on
+        """Create a contour plot of a specified quantity.
 
         Parameters
         ----------
-        qty_str : str, optional
-            Quantity to plot. Default `"RealizedGain"`.
+        farfield_quantity : str, optional
+            Far field quantity to plot. The default is ``"RealizedGain"``.
+            Available functions are: ``"RealizedGain"``, ``"RealizedGain_Phi"``, ``"RealizedGain_Theta"``,
+            ``"rEPhi"``, ``"rETheta"``, and ``"rETotal"``.
         phi_scan : float, int, optional
-            Phi Scan Angle in degree. Default `0`.
+            Phi scan angle in degrees. The default is ``0``.
         theta_scan : float, int, optional
-            Theta Scan Angle in degree. Default `0`.
+            Theta scan angle in degrees. The default is ``0``.
         title : str, optional
-            Plot title. Default `"RectangularPlot"`.
-        convert_to_db : bool, optional
-            Either if the quantity has to be converted in db or not. Default is `True`.
+            Plot title. The default is ``"RectangularPlot"``.
+        format_quantity : str, optional
+            Conversion data function.
+            Available functions are: ``"abs"``, ``"ang"``, ``"dB10"``, ``"dB20"``, ``"deg"``, ``"imag"``, ``"norm"``,
+            and ``"real"``.
         export_image_path : str, optional
-            Full path to image file. Default is None to not export.
+            Full path for the image file. The default is ``None``, in which case the file is not exported.
+        levels : int, optional
+                Colormap levels. The default is ``64``.
+        show : bool, optional
+            Whether to show the plot. The default is ``True``. If ``False``, the Matplotlib
+            instance of the plot is shown.
 
         Returns
         -------
         :class:`matplotlib.plt`
-            Matplotlib fig object.
+            If ``show=True``, it returns a Matplotlib figure instance of the plot.
+            If ``show=False``, it returns the plotted curve.
+
```

Comment:
```suggestion
            If ``show=True``, a Matplotlib figure instance of the plot is returned.
            If ``show=False``, the plotted curve is returned.

```
This seems to contradict the description for the "show" parameter. Also, in that description, I'd use "Whether to show the plotted curve." This way, it matches the description here in the "Returns" section.